incubator/exporter-prometheus/src/main/java/io/aklivity/zilla/runtime/exporter/prometheus/internal/PrometheusExporterHandler.java

```diff
@@ -0,0 +1,173 @@
+import io.aklivity.zilla.runtime.exporter.prometheus.internal.processor.MetricsProcessor;
+
+public class PrometheusExporterHandler implements ExporterHandler
+{
+    private final EndpointConfig endpoint;
+    private final LayoutManager manager;
+    private final PrometheusMetricDescriptor metricDescriptor;
+
+    private Map<Metric.Kind, List<MetricsLayout>> layouts;
+    private MetricsProcessor metrics;
+    private HttpServer server;
+
+    public PrometheusExporterHandler(
+        EngineConfiguration config,
+        ExporterConfig exporter)
+    {
+        this.endpoint = resolveEndpoint(exporter);
+        this.manager = new LayoutManager(config.directory());
+        this.metricDescriptor = new PrometheusMetricDescriptor(config);
+    }
+
+    @Override
+    public void start()
+    {
+        try
+        {
+            layouts = Map.of(
+                COUNTER, manager.countersLayouts(),
+                GAUGE, manager.gaugesLayouts(),
+                HISTOGRAM, manager.histogramsLayouts()
+            );
+            metrics = new MetricsProcessor(layouts, manager.labels(), metricDescriptor::kind, metricDescriptor::name,
+                metricDescriptor::description, null, null);
+            server = HttpServer.create(new InetSocketAddress(endpoint.port), 0);
+            server.createContext(endpoint.path, new MetricsHttpHandler());
+            server.start();
+        }
+        catch (IOException ex)
+        {
+            LangUtil.rethrowUnchecked(ex);
+        }
+    }
+
+    @Override
+    public int export()
+    {
+        return 0;
+    }
+
+    @Override
+    public void stop()
+    {
+        if (server != null)
+        {
+            server.stop(0);
+            server = null;
+        }
+        layouts.keySet().stream().flatMap(kind -> layouts.get(kind).stream()).forEach(MetricsLayout::close);
+    }
+
+    private EndpointConfig resolveEndpoint(
+        ExporterConfig exporter)
+    {
+        PrometheusOptionsConfig options = (PrometheusOptionsConfig) exporter.options;
+        EndpointConfig endpoint;
+
+        if (options != null && options.endpoints != null && options.endpoints.length > 0)
+        {
+            endpoint = options.endpoints[0];
+            if (endpoint.scheme == null)
+            {
+                endpoint.scheme = DEFAULT.scheme;
+            }
+            if (endpoint.port == 0)
+            {
+                endpoint.port = DEFAULT.port;
+            }
+            if (endpoint.path == null)
+            {
+                endpoint.path = DEFAULT.path;
+            }
+        }
+        else
+        {
+            endpoint = DEFAULT;
+        }
+        return endpoint;
+    }
+
+    private String generateOutput()
+    {
+        String output = "";
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        PrintStream out = new PrintStream(os);
+        metrics.print(out);
+        try
+        {
+            output = os.toString("UTF8");
```

Comment:
This is triggering plenty of object allocation, so we'll take another pass on that after these changes.